File: 🟡 Medium/3163 - String Compression III/_3163StringCompressionIII.py
class Solution:
    def compressedString(self, word: str) -> str:
        res = ""
        i = 0
        curr = word[0]

        while i < len(word):
            count = 0

            while i < len(word) and word[i] == curr and count < 9:
                i += 1
                count += 1

            res += str(count) + curr

            if i != len(word):
                curr = word[i]

        return res

File: 🟡 Medium/3163 - String Compression III/_3163StringCompressionIII.cpp
class Solution {
public:
    string compressedString(string word) {
        string res = "";
        int i = 0;
        
        char curr = word[0];

        while (i < word.size()) {
            int count = 0;

            while (i < word.size() && word[i] == curr && count < 9) {
                i++;
                count++;
            }

            res += to_string(count) + curr;

            if (i != word.size()) {
                curr = word[i];
            }
        }

        return res;
    }
};